django/core/paginator.py

```diff
@@ -236,3 +356,96 @@ def end_index(self):
         if self.number == self.paginator.num_pages:
             return self.paginator.count
         return self.number * self.paginator.per_page
+
+
+class AsyncPage:
+    def __init__(self, object_list, number, paginator):
+        self.object_list = object_list
+        self.number = number
+        self.paginator = paginator
+
+    def __repr__(self):
+        return "<Async Page %s>" % self.number
+
+    async def __aiter__(self):
+        if hasattr(self.object_list, "__aiter__"):
+            async for obj in self.object_list:
+                yield obj
+        else:
+            for obj in self.object_list:
+                yield obj
```

Comment:
```suggestion
    async def __aiter__(self):
        for obj in self.object_list:
            yield obj
```
I can update this to the above without any test failures. Can you add a test to cover the `__aiter__` case?

Reply:
Yes, I will add it!